src/sprout/Controllers/DbToolsController.php

```diff
@@ -1460,6 +1468,164 @@ public function exportFilesAction()
     }
 
 
+    /**
+    * Simple test tools for working out Open AI tooling using the client's keys
+    **/
+    public function openAiTest()
+    {
+        $key = Kohana::config('openai.secret_key');
+        if (empty($key)) {
+            echo '<p><em>OpenAI secret key not configured in config.openai.secret_key</em></p>';
+            $this->template('AI Tooling');
+            return;
+        }
+
+        $view = new PhpView('sprout/dbtools/openai_test');
+        echo $view->render();
+
+        $this->template('AI Tooling');
+    }
+
+
+    /**
+     * Pass a request to Open AI API and render out the result accordingly
+     *
+     * @return void
+     */
+    public function openAiTestSubmit()
+    {
+        Csrf::checkOrDie();
+
+        $endpoint = $_POST['endpoint'];
+
+        if (!method_exists('Sprout\\Helpers\\AI\\OpenAiApi', $endpoint)) {
+            $endpoint = Enc::html($endpoint);
+            echo "<p><em>Invalid endpoint: {$endpoint}</em></p>";
+            return;
+        }
+
+        $response = OpenAiApi::$endpoint($_POST['prompt']);
+
+        // Setting to output debug data ('', 'tokens' or 'debug')
+        $debug = $_POST['debug_data'];
+
+        match ($endpoint) {
+            'chatCompletion' => $this->openAiTextOutput($response, $debug),
+            'imageGenerateSrc' => $this->openAiImageOutputSrc($response, $debug),
+            'imageGenerateBlob' => $this->openAiImageOutputBlob($response, $debug),
+            default => throw new \Exception('Unsupported endpoint'),
+        };
+    }
+
+
+    /**
+     * Render out a text response
+     *
+     * @param string $response
+     * @param string $debug Setting to output debug data ('', 'tokens' or 'debug')
+     *
+     * @return void
+     */
+    public function openAiTextOutput(string $response, string $debug): void
+    {
+        echo $response;
+        $this->openAiDebugOutput($debug);
+    }
+
+
+    /**
+     * Render out n image URL response
+     *
+     * @param string $response
+     * @param string $debug Setting to output debug data ('', 'tokens' or 'debug')
+     *
+     * @return void
+     */
+    public function openAiImageOutputSrc(string $response, string $debug): void
+    {
+        echo '<img src="', $response, '" style="max-width: 100%">';
+        $this->openAiDebugOutput($debug);
+    }
+
+
+    /**
+     * Render out n image blob response
+     *
+     * @param string $response
+     * @param string $debug Setting to output debug data ('', 'tokens' or 'debug')
+     *
+     * @return void
+     */
+    public function openAiImageOutputBlob(string $response, string $debug): void
+    {
+        echo "<img src=\"data:image/jpeg;base64,{$response}\" style=\"max-width: 100%\">";
```

Comment:
Same here, anything to quickly validate base64 blobs?

like a silly `/^[0-9a-f=]*$/i` regex or something.